### PR TITLE
Adding Ludo's Model

### PR DESCRIPTION
**Pull request by bischtob**
This PR adds scripts for Ludo's model, both analysis and simulation.

- [x] Added new Right-hand sides
- [x] Added periodic BCs for Ludo model
- [x] Added runscripts
- [x] Added Ludo's changes

### CODE REVIEW

File: examples/grf/rhs.jl [hunk 1]
```diff
@@ -65,6 +76,56 @@ function make_deterministic_tendency(model::LinearDiffusionSDE{FT, CM, Dirichlet
...
+                k = (j-1)*N+i
+                k_ip1 = k+1
+                k_im1 = k-1
+                k_jp1 = k + N
+                k_jm1 = k - N
+                # At the boundary, we have u_bc on the face
+                # while u[k] is at the cell center. -> Δx -> Δx/2
+                if i == N
+                    v_bc = v[(j-1)*N+1]
+                    du[k] += ((v[k_im1] - v[k]) - (v[k] - v_bc)/2)
```
Comment (kmdeck) [inline examples/grf/rhs.jl]: I think for periodic BC we do not need the factor of 2. when we supply dirichlet BC, the value is given on the face, so the distance between the center and face is Delta/2. when we use periodic BC, the value is at the adjacent/wrapped cell center, so distance shouldnt be Delta?

Comment (kmdeck) [reply examples/grf/rhs.jl]: For reference:
i = j = 1 upper left corner, i = 1, j = N is the upper right corner, i = N, j = 1: lower left corner, i=j=N is the lower right corner

File: examples/grf/sim_ludo_analysis.jl [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+# Boundary condition - periodic
+Γ = FT.(reshape(zeros(N^4), (N^2,N^2)))
+for i1 in 1:N
+    for j1 in 1:N
+        k1 = (j1-1)*N+i1
+        for i2 in 1:N
+            for j2 in 1:N
+                k2 = (j2-1)*N+i2
+                Γ[k1,k2] = FT(1.0/sqrt((i1-i2)^2 + (j1-j2)^2+1))
```
Comment (kmdeck) [inline examples/grf/sim_ludo_analysis.jl]: double check this is the same as what Ludo used?

File: examples/grf/sim_ludo.jl [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+# Allocate noise vectors up front.
+W = zeros(FT, N*N)
+W_corr = similar(W)
+model = LudoDiffusionSDE(σ, α, β, γ, N, ΓL, W, W_corr)
+
+deterministic_tendency! = make_deterministic_tendency(model)
+stochastic_increment! = make_stochastic_increment(model)
+
+# Initial condition
+u = FT.(zeros(N^2));
```
Comment (kmdeck) [inline examples/grf/sim_ludo.jl]: could try random values between -1 and 1

File: examples/grf/rhs.jl (hunk 1 above)
Comment (bischtob) [reply examples/grf/rhs.jl]: Please double check again :)

File: examples/grf/sim_ludo_analysis.jl (hunk 2 above)
Comment (bischtob) [reply examples/grf/sim_ludo_analysis.jl]: Modified to Ludo's code

File: examples/grf/sim_ludo.jl (hunk 3 above)
Comment (bischtob) [reply examples/grf/sim_ludo.jl]: Done.

File: examples/grf/sim_ludo_analysis.jl [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+@time for i in ProgressBar(1:nsteps)
+    t = tspan[1]+dt*(i-1)
+    Euler_Maruyama_step!(du, u, t, deterministic_tendency!, stochastic_increment!, dt)
+    if i ∈ savesteps
+        save_index = Int(i/n_steps_per_save)
+        solution[:,:, save_index] .= reshape(u, (N,N))
+    end
+end
+
+# spinup should be zero
```
Comment (kmdeck) [inline examples/grf/sim_ludo_analysis.jl:87]: still true for this system?

File: examples/grf/sim_ludo.jl [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+deterministic_tendency! = make_deterministic_tendency(model)
+stochastic_increment! = make_stochastic_increment(model)
+
+# Initial condition
+u = 2*rand(FT, N^2).-1;
+
+# Preallocate
+du = similar(u);
+# Integration time, timestep, nsteps
```
Comment (kmdeck) [inline examples/grf/sim_ludo.jl:65]: I assume 100 is the autocorrelation time? worth noting?

Comment (kmdeck) [reply examples/grf/sim_ludo.jl:65]: is it the same regardless of parameters used?

**Review by kmdeck — APPROVED** [review]
Looks good Toby! I removed the regularization since we didnt end up using that and clarified a couple comments.

only remaining ?s are:
- is 100 the autocorrelation time regardless of size N and other parameters?
- do we need to subtract spinup?
- could you squash before commiting?

File: examples/grf/sim_ludo.jl (hunk 5 above)
Comment (bischtob) [reply examples/grf/sim_ludo.jl:65]: yes, it's the same on. I shared some plots in Slack that show the autocorrelation time doesn't really depend on the resolution.

**Pull request merged**